### PR TITLE
Fix 4009 add_layers from catalog

### PR DESCRIPTION
**Pull request by MV88**
## Description
This pr add the possibility to add layers from url by specifying the layername with workspace and the name of the catalog present in map.

Note that this pr is incomplete, it needs: 
- [x] Manual Test wmts layer
~Manual Test esri layer~
- [x] All the needed unit tests

## Issues
 - #4009

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Feature

**What is the current behavior?** (You can also link to an open issue here)
you cannot dinamycally add layer on map loading

**What is the new behavior?**
you can add lyer from map viewer query param

**Does this PR introduce a breaking change?** (check one with "x", remove the other)

 - [ ] Yes, and I documented them in migration notes
 - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

```
here are some tests url using it under genova project

// ONE LAYER, CSW CATALOG
http://localhost:8081/?debug=true#/viewer/openlayers/new?actions=[{"type":"CATALOG:ADD_LAYERS_MAPVIEWER_URL","layers":["DBT:SOTTOPASSO"],"sources":["GeoPortale Comune di Genova"]}]

// TWO LAYERS, SAME CSW CATALOG

http://localhost:8081/?debug=true#/viewer/openlayers/new?actions=[{"type":"CATALOG:ADD_LAYERS_MAPVIEWER_URL","layers":["DBT:SOTTOPASSO", "PROTCIV:INCENDI_INTERFACCIA"],"sources":["GeoPortale Comune di Genova", "GeoPortale Comune di Genova"]}]

// TWO LAYERS, SAME CSW CATALOG, DIFFERENT ORDER
http://localhost:8081/?debug=true#/viewer/openlayers/new?actions=[{"type":"CATALOG:ADD_LAYERS_MAPVIEWER_URL","layers":["PROTCIV:INCENDI_INTERFACCIA","DBT:SOTTOPASSO"],"sources":["GeoPortale Comune di Genova", "GeoPortale Comune di Genova"]}]


// ERROR with wrong layer name 
http://localhost:8081/?debug=true#/viewer/openlayers/new?actions=[{"type":"CATALOG:ADD_LAYERS_MAPVIEWER_URL","layers":["error","DBT:SOTTOPASSO"],"sources":["GeoPortale%20Comune%20di%20Genova",%20"GeoPortale%20Comune%20di%20Genova"]}]
```

### CODE REVIEW

File: web/client/test-resources/csw/getRecordsResponse-gs-us_statesWMTS.xml [hunk 1]
```diff
@@ -0,0 +1,8194 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<Capabilities xmlns="http://www.opengis.net/wmts/1.0" xmlns:ows="http://www.opengis.net/ows/1.1" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:gml="http://www.opengis.net/gml" xsi:schemaLocation="http://www.opengis.net/wmts/1.0 http://schemas.opengis.net/wmts/1.0/wmtsGetCapabilities_response.xsd" version="1.0.0">
```
Comment (offtherailz) [inline web/client/test-resources/csw/getRecordsResponse-gs-us_statesWMTS.xml]: Reuse existing test data

File: web/client/test-resources/csw/getRecordsResponse-gs-us_states.xml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8"?><csw:GetRecordsResponse xmlns:csw="http://www.opengis.net/cat/csw/2.0.2" xmlns:rim="urn:oasis:names:tc:ebxml-regrep:xsd:rim:3.0" xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:dct="http://purl.org/dc/terms/" xmlns:ows="http://www.opengis.net/ows" xmlns:ogc="http://www.opengis.net/ogc" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" version="2.0.2" xsi:schemaLocation="http://www.opengis.net/cat/csw/2.0.2 https://gs-stable.geo-solutions.it/geoserver/schemas/csw/2.0.2/record.xsd">
+  <csw:SearchStatus timestamp="2019-08-13T16:03:43.292Z"/>
```
Comment (offtherailz) [inline web/client/test-resources/csw/getRecordsResponse-gs-us_states.xml]: Reuse existing test data

File: web/client/actions/catalog.js [hunk 3]
```diff
@@ -46,6 +51,39 @@ export const TOGGLE_TEMPLATE = 'CATALOG:TOGGLE_TEMPLATE';
...
 export const TOGGLE_ADVANCED_SETTINGS = 'CATALOG:TOGGLE_ADVANCED_SETTINGS';
 
+/**
+ * it adds a list of layers given its catalog to the map
+ * @param {string[]} layers list with workspace to be added in the map
+ * @param {string[]} sources catalog names related to each layer
+ */
+export function addLayersMapViewerUrl(layers = [], sources = []) {
+    return {
+        type: ADD_LAYERS_MAPVIEWER_URL,
```
[the remark this answers is not in the document]
Comment (MV88) [reply web/client/actions/catalog.js]: i would go for:
addLayersFromCatalogs
ADD_LAYERS_FROM_CATALOGS

File: web/client/test-resources/csw/getRecordsResponse-gs-us_states.xml (hunk 2 above)
Comment (MV88) [reply web/client/test-resources/csw/getRecordsResponse-gs-us_states.xml]: in this case i want only one result.

File: web/client/test-resources/csw/getRecordsResponse-gs-us_statesWMTS.xml (hunk 1 above)
Comment (MV88) [reply web/client/test-resources/csw/getRecordsResponse-gs-us_statesWMTS.xml]: i'll try to see if there is something usable

File: web/client/actions/catalog.js [hunk 4]
```diff
@@ -46,6 +51,39 @@ export const TOGGLE_TEMPLATE = 'CATALOG:TOGGLE_TEMPLATE';
...
+ */
+export function addLayersMapViewerUrl(layers = [], sources = []) {
+    return {
+        type: ADD_LAYERS_FROM_CATALOGS,
+        layers,
+        sources
+    };
+}
+/**
+ * it search for a layer in the related catalog
```
Comment (offtherailz) [inline web/client/actions/catalog.js]:
```suggestion
 * Searches for a layer in the related catalog
```

File: web/client/test-resources/csw/getRecordsResponse-gs-us_states.xml (hunk 2 above)
Comment (offtherailz) [reply web/client/test-resources/csw/getRecordsResponse-gs-us_states.xml]: so remove the references to gs-stable,geo-solutions.it

**Pull request merged**